File: medium/82. Remove Duplicates from Sorted List II/0208.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        fast = head
        dummy = ListNode(0, head)
        slow = dummy
        while fast:
            if fast.next and fast.val==fast.next.val:
                while fast.next and fast.val==fast.next.val:
                    fast=fast.next
                slow.next=fast.next
            else:
                slow=slow.next
            fast=fast.next
        return dummy.next

File: medium/82. Remove Duplicates from Sorted List II/0208.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
    let fast = head;
    let slow = new ListNode(0, head);
    let prev = slow;
    while (fast) {
        if (fast.next && fast.val == fast.next.val) {
            while (fast.next && fast.val == fast.next.val) {
                fast = fast.next
            }
            prev.next = fast.next
        } else {
            prev = prev.next
        }
        fast = fast.next
    }
    return slow.next
};